### PR TITLE
src/components: add FormRegister component with component/e2e tests

### PR DESCRIPTION
**Pull request by maceksimon**
* Add `RegisterPage` and `/register` route
* Add `FormRegister` component with email, password and password confirm
* Add translations
* Add tests

### CODE REVIEW

File: src/components/FormRegister.vue [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    const formRegister = reactive({
+      email: '',
+      password: '',
+      passwordConfirm: '',
+    });
+
+    const isPassword = ref(true);
+    const isPasswordConfirm = ref(true);
+
+    const isValid = (val: string): boolean => val?.length > 0;
```
Comment (tmszi) [inline src/components/FormRegister.vue:40]: We can [reuse this func](https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/fd447cbacbbeac6f08fe421cdfee8800f35d73d6/src/components/FormLogin.vue#L63) and func should be in ./src/composables/.

File: src/components/FormRegister.vue [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    const isEmail = (value: string): boolean => {
```
Comment (tmszi) [inline src/components/FormRegister.vue:42]: Same as my previous comment https://github.com/auto-mat/ride-to-work-by-bike-frontend/pull/111/files#r1384364179

File: src/components/__tests__/FormRegister.cy.js [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        .find('.q-field__control')
+        .should('be.visible')
+        .and('have.css', 'border-radius', '8px');
+    });
+
+    it('renders show/hide icons for password inputs', () => {
+      // password
+      cy.dataCy('form-register-password-icon')
+        .should('contain', 'visibility')
+        .should('have.color', `${colorPrimary}`);
```
Comment (tmszi) [inline src/components/__tests__/FormRegister.cy.js:89]: `.and()`

File: src/components/__tests__/FormRegister.cy.js [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+      cy.dataCy('form-register-password-icon')
+        .invoke('height')
+        .should('be.equal', 18);
+      cy.dataCy('form-register-password-icon')
+        .invoke('width')
+        .should('be.equal', 18);
+      // password confirm
+      cy.dataCy('form-register-password-confirm-icon')
+        .should('contain', 'visibility')
+        .should('have.color', `${colorPrimary}`);
```
Comment (tmszi) [inline src/components/__tests__/FormRegister.cy.js:99]: `.and()`

File: src/components/FormRegister.vue [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        </label>
+        <!-- Input -->
+        <q-input
+          dense
+          outlined
+          hide-bottom-space
+          bg-color="grey-1"
+          v-model="formRegister.passwordConfirm"
+          id="form-register-password"
+          :type="isPasswordConfirm ? 'password' : 'text'"
```
Comment (tmszi) [inline src/components/FormRegister.vue:175]: Maybe we can use `lazy-rules` validation props.

File: src/components/FormRegister.vue [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        <!-- Input -->
+        <q-input
+          dense
+          outlined
+          hide-bottom-space
+          bg-color="grey-1"
+          v-model="formRegister.password"
+          id="form-register-password"
+          :hint="$t('register.form.hintPassword')"
+          :type="isPassword ? 'password' : 'text'"
```
Comment (tmszi) [inline src/components/FormRegister.vue:136]: Maybe we can use `lazy-rules` validation props.

File: src/router/routes.ts [hunk 7]
```diff
@@ -11,6 +11,11 @@ const routes: RouteRecordRaw[] = [
     component: () => import('layouts/LoginLayout.vue'),
     children: [{ path: '', component: () => import('pages/LoginPage.vue') }],
   },
+  {
+    path: '/register',
+    component: () => import('layouts/LoginLayout.vue'),
```
Comment (tmszi) [inline src/router/routes.ts:16]: If we use `LoginLayout` component for registration page component, layout component file and component name should be `RegisterLoginLayout.vue` and you can update component doc header, please.

https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/fd447cbacbbeac6f08fe421cdfee8800f35d73d6/src/layouts/LoginLayout.vue#L2-L26

File: src/components/FormRegister.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/FormRegister.vue]: Yes. Extracted all validation functions into a composable.

File: src/components/__tests__/FormRegister.cy.js (hunk 3 above)
Comment (maceksimon) [reply src/components/__tests__/FormRegister.cy.js:89]: Fixed.

File: src/components/__tests__/FormRegister.cy.js (hunk 4 above)
Comment (maceksimon) [reply src/components/__tests__/FormRegister.cy.js:99]: Fixed.

File: src/components/FormRegister.vue (hunk 5 above)
Comment (maceksimon) [reply src/components/FormRegister.vue:157]: Added.

File: src/components/FormRegister.vue (hunk 6 above)
Comment (maceksimon) [reply src/components/FormRegister.vue:116]: Added.

File: src/router/routes.ts (hunk 7 above)
Comment (maceksimon) [reply src/router/routes.ts]: Absolutely. Updated.

**Review by tmszi — APPROVED** [review]

**Pull request merged**